src/Http/Routing/src/DependencyInjection/RoutingServiceCollectionExtensions.cs

```diff
@@ -26,6 +27,22 @@ public static class RoutingServiceCollectionExtensions
     /// <param name="services">The <see cref="IServiceCollection"/> to add the services to.</param>
     /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
     public static IServiceCollection AddRouting(this IServiceCollection services)
+    {
+        return services.AddRoutingCore().Configure<RouteOptions>(options =>
+        {
+            // Replace RegexErrorStubRouteConstraint with the real regex constraint.
+            options.SetParameterPolicy<RegexInlineRouteConstraint>("regex");
```

Comment:
What if someone already already customized the "regex" constraint? Can we verify that the `RegexErrorStubRouteConstraint` is still there before replacing it? It might not be something that happens much if at all in practice, but it seems easy enough to check for and test using internal APIs.